### PR TITLE
Add dataset env var implementaion

### PR DESCRIPTION
**Pull request by rmn-boiko**
## Description

Closes: https://github.com/kamu-data/kamu-node/issues/56

## Checklist before requesting a review

- [x] Unit and integration tests added
  
- [x] Compatibility:
    
  - [x] Network APIs: ✅
    
  - [x] Workspace layout and metadata: ✅ 
    
  - [x] Configuration: ✅
    
  - [x] Container images: ✅
- [x] Documentation:
    
  - [x] [Changelog](./CHANGELOG.md): ✅
    
  - [x] [Public documentation](https://github.com/kamu-data/kamu-docs/): ✅
  
- [ ] Downstream effects:
    
  - [ ] [kamu-node](https://github.com/kamu-data/kamu-node): ✅

### CODE REVIEW

File: src/adapter/graphql/src/mutations/dataset_env_vars_mut.rs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Interface, Debug, Clone)]
+#[graphql(field(name = "message", ty = "String"))]
+pub enum ModifyDatasetEnvVarResult {
+    Success(ModifyDatasetEnvVarResultSuccess),
+    NotFount(ModifyDatasetEnvVarResultNotFound),
```
Comment (zaychenko-sergei) [inline src/adapter/graphql/src/mutations/dataset_env_vars_mut.rs:231]: founD

File: src/adapter/graphql/src/queries/datasets/dataset_env_var.rs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        self.env_var.key.clone()
+    }
+
+    /// Value of dataset environment variable
+    #[allow(clippy::unused_async)]
+    async fn value(&self) -> Result<String> {
+        self.env_var
+            .get_exposed_value()
+            .map_err(|err| GqlError::Internal(err.int_err()))
+    }
```
Comment (zaychenko-sergei) [inline src/adapter/graphql/src/queries/datasets/dataset_env_var.rs:45]: Maybe we shoudln't show the secret value until explicitly requested?

File: src/adapter/graphql/src/queries/datasets/dataset.rs [hunk 3]
```diff
@@ -98,6 +98,11 @@ impl Dataset {
         DatasetMetadata::new(self.dataset_handle.clone())
     }
 
+    /// Access to the environment variable of this dataset
+    async fn env_vars(&self) -> DatasetEnvVars {
+        DatasetEnvVars::new(self.dataset_handle.clone())
```
Comment (zaychenko-sergei) [inline src/adapter/graphql/src/queries/datasets/dataset.rs:103]: Don't we need to check read permissions?
Also, to view secrets, I think we need a separate permission.

File: Cargo.toml [hunk 4]
```diff
@@ -17,11 +17,13 @@ members = [
     # Domain
     "src/domain/opendatafabric",
     "src/domain/accounts/domain",
+    "src/domain/dataset-env-vars/domain",
```
Comment (zaychenko-sergei) [inline Cargo.toml:20]: Maybe a bit too narrow name for a domain. Sooner or later we will have dataset entity present in the database, and other connected data, like permissions for a private datasets. Other features would follow. I would reserve a wider name, maybe `src/domain/dataset`

File: src/domain/dataset-env-vars/domain/src/repos/dataset_env_var_repository.rs [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        secret_nonce: Option<Vec<u8>>,
+    ) -> Result<(), ModifyDatasetEnvVarError>;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct DatasetEnvVarPaginationOpts {
+    pub limit: i64,
+    pub offset: i64,
+}
```
Comment (zaychenko-sergei) [inline src/domain/dataset-env-vars/domain/src/repos/dataset_env_var_repository.rs:66]: This isn't specific to env vars. Could be useful for reuse in `database-common`. Look for similar structs and replace them too - I guess we had one in access tokens topic.

File: src/domain/dataset-env-vars/services/src/dataset_env_var_service_static_impl.rs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    #[allow(clippy::needless_pass_by_value)]
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+impl DatasetEnvVarService for DatasetEnvVarServiceStaticImpl {
```
Comment (zaychenko-sergei) [inline src/domain/dataset-env-vars/services/src/dataset_env_var_service_static_impl.rs:45]: What's the use case for this implementation?

File: src/domain/opendatafabric/src/identity/dataset_id.rs [hunk 7]
```diff
@@ -204,3 +204,56 @@ impl<'de> serde::de::Visitor<'de> for DatasetIDSerdeVisitor {
...
 }
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/// SQLX
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[cfg(feature = "sqlx")]
+impl<'r, DB: sqlx::Database> sqlx::Decode<'r, DB> for DatasetID
+where
+    &'r str: sqlx::Decode<'r, DB>,
```
Comment (zaychenko-sergei) [inline src/domain/opendatafabric/src/identity/dataset_id.rs:215]: I guess this part was copypasted from `AccountID`. Should we have a macro for such newtypes?

File: src/domain/dataset-env-vars/services/src/dataset_env_var_service_static_impl.rs (hunk 6 above)
Comment (zaychenko-sergei) [reply src/domain/dataset-env-vars/services/src/dataset_env_var_service_static_impl.rs:45]: If it's just tests, why would we care to have configuration for this?

File: src/adapter/graphql/src/queries/datasets/dataset.rs (hunk 3 above)
Comment (rmn-boiko) [reply src/adapter/graphql/src/queries/datasets/dataset.rs:103]: read permission check is inside `dataset_env_vars.rs`

**Review by zaychenko-sergei — APPROVED** [review]
Changelog would be nice

**Pull request merged**